### PR TITLE
Don't offer FixReturnType fix if both are void

### LINKED ISSUE #47089: Fix return type is incorrectly offered for void expression
**Version Used**: VS 16.8.0 Preview 1.0 (C# Tools   3.8.0-1.20378.6+226c0c3f392232b3dd7ecd527f1fd00f0f86d7e4)

**Steps to Reproduce**:

1. Write the following incorrect code:

    ```c#
    void f()
    {
        return Console.WriteLine();
    }
    ```
2. Use <kbd>Ctrl</kbd>+<kbd>.</kbd> on the `return` keyword and select Fix return type.

**Expected Behavior**: Fix return type is not offered. Some valid fix might be offered instead, possibly changing `return expr;` to `expr; return;`.

**Actual Behavior**: Fix return type is offered and does nothing:

### PR DESCRIPTION
**Pull request by Youssef1313**
Fixes #47089

### CODE REVIEW

File: src/EditorFeatures/CSharpTest/Diagnostics/FixReturnType/FixReturnTypeTests.cs [hunk 1]
```diff
@@ -286,6 +286,19 @@ public async Task ExpressionBodied()
 @"class C
 {
     int M() => 1[||];
+}");
+        }
+
+        [Fact]
```
Comment (CyrusNajmabadi) [inline src/EditorFeatures/CSharpTest/Diagnostics/FixReturnType/FixReturnTypeTests.cs:292]: Add workitem

File: src/Features/CSharp/Portable/CodeFixes/FixReturnType/CSharpFixReturnTypeCodeFixProvider.cs [hunk 2]
```diff
@@ -50,14 +50,17 @@ public override async Task RegisterCodeFixesAsync(CodeFixContext context)
             var cancellationToken = context.CancellationToken;
 
             var analyzedTypes = await TryGetOldAndNewReturnTypeAsync(document, diagnostics, cancellationToken).ConfigureAwait(false);
-            if (analyzedTypes == default)
+            if (analyzedTypes == default ||
+                (IsVoid(analyzedTypes.declarationToFix) && IsVoid(analyzedTypes.fixedDeclaration)))
```
Comment (CyrusNajmabadi) [inline src/Features/CSharp/Portable/CodeFixes/FixReturnType/CSharpFixReturnTypeCodeFixProvider.cs:54]: Too complex. Please make into separate checks (and add comment please)

File: src/Features/CSharp/Portable/CodeFixes/FixReturnType/CSharpFixReturnTypeCodeFixProvider.cs [hunk 3]
```diff
@@ -50,14 +50,17 @@ public override async Task RegisterCodeFixesAsync(CodeFixContext context)
...
             {
                 return;
             }
 
             context.RegisterCodeFix(
                new MyCodeAction(c => FixAsync(document, diagnostics.First(), c)),
                diagnostics);
+
+            static bool IsVoid(TypeSyntax typeSyntax) => typeSyntax is PredefinedTypeSyntax predefined && predefined.Keyword.IsKind(SyntaxKind.VoidKeyword);
```
Comment (CyrusNajmabadi) [inline src/Features/CSharp/Portable/CodeFixes/FixReturnType/CSharpFixReturnTypeCodeFixProvider.cs:63]: Please put `=>` on next line. Please add an explicit `return` statement above all local functions

File: src/EditorFeatures/CSharpTest/Diagnostics/FixReturnType/FixReturnTypeTests.cs (hunk 1 above)
Comment (Youssef1313) [inline src/EditorFeatures/CSharpTest/Diagnostics/FixReturnType/FixReturnTypeTests.cs:292]:
```suggestion
        [Fact]
        [WorkItem(47089, "https://github.com/dotnet/roslyn/issues/47089")]
```

**Comment by Youssef1313** [comment]
@CyrusNajmabadi Thanks for review. I addressed your suggested changes.

**Comment by Youssef1313** [comment]
@CyrusNajmabadi The build is complete now.

**Comment by CyrusNajmabadi** [comment]
@Youssef1313 is there an equivalent problem on the VB side?

**Comment by Youssef1313** [comment]
> @Youssef1313 is there an equivalent problem on the VB side?

I missed that. There is a parallel VB code fix [here](https://github.com/dotnet/roslyn/blob/31ebe4954e5455e630acb1e8445ecb742c678753/src/Features/VisualBasic/Portable/CodeFixes/IncorrectFunctionReturnType/IncorrectFunctionReturnTypeCodeFixProvider.vb). I haven't checked yet if it has the same issue or not.

**Pull request merged**